pkg/sloop/storemanager/storemanager.go

```diff
@@ -76,21 +80,46 @@ func (sm *StoreManager) Start() {
 				glog.Infof("Store manager main loop exiting")
 				return
 			}
-			temporaryEmitMetrics(sm.storeRoot, sm.tables.Db(), sm.fs)
+			temporaryEmitMetrics(sm.config.StoreRoot, sm.tables.Db(), sm.fs)
 			metricGcRunCount.Inc()
-			cleanupPerformed, err := doCleanup(sm.tables, sm.storeRoot, sm.timeLimit, sm.sizeLimitMb*1024*1024, sm.fs)
+			cleanupPerformed, err := doCleanup(sm.tables, sm.config.StoreRoot, sm.config.TimeLimit, sm.config.SizeLimitMb*1024*1024, sm.fs)
 			if err != nil {
-				glog.Errorf("GC failed with err:%v, will sleep: %v and retry later ...", err, sm.freq)
-				sm.sleeper.Sleep(sm.freq)
+				glog.Errorf("GC failed with err:%v, will sleep: %v and retry later ...", err, sm.config.Freq)
 			} else if !cleanupPerformed {
-				glog.V(2).Infof("GC did not need to clean anything, will sleep: %v", sm.freq)
-				sm.sleeper.Sleep(sm.freq)
+				glog.V(2).Infof("GC did not need to clean anything, will sleep: %v", sm.config.Freq)
 			} else {
 				// We did some cleanup and there were no errors.  Because we may be in a low space situation lets skip
 				// the sleep and repeat the loop
-				glog.Infof("GC cleanup performed")
+				glog.Infof("GC cleanup performed without errors, next run: %v", sm.config.Freq)
 				metricGcCleanupPerformedCount.Inc()
 			}
+			sm.sleeper.Sleep(sm.config.Freq)
+		}
+	}()
+
+	// Its up to us to trigger the Badger value log GC.
+	// See https://github.com/dgraph-io/badger#garbage-collection
+	go func() {
+		if sm.config.BadgerVLogGCFreq == 0 {
+			return
+		}
+		sm.wg.Add(1)
+		defer sm.wg.Done()
+		for {
+			if sm.isDone() {
+				glog.Infof("ValueLogGC loop exiting")
+				return
+			}
+
+			for {
+				before := time.Now()
+				err := sm.tables.Db().RunValueLogGC(sm.config.BadgerDiscardRatio)
```

Comment:
should there be a metric published that this occurred?

Reply:
Added